src/gordon_gcp/plugins/enricher.py

```diff
@@ -49,11 +56,87 @@ class GCEEnricher:
     """
     phase = 'enrich'
 
-    def __init__(self, config, success_channel, error_channel):
+    def __init__(self, config, http_client, success_channel, error_channel):
         self.config = config
         self.success_channel = success_channel
         self.error_channel = error_channel
+        self._http_client = http_client
+
+    async def _poll_for_ext_ip(self, resource_name):
+        exception = None
+        backoff = 2
+        base_url = f'https://compute.googleapis.com/compute/v1/{resource_name}'
+
+        for attempt in range(1, self.config['retries'] + 1):
+            try:
+                logging.debug(f'Attempt {attempt}: fetching {base_url}')
+                instance_data = await self._http_client.get_json(base_url)
+                return instance_data['networkInterfaces'][0][
+                    'accessConfigs'][0]['natIP']
+            except exceptions.GCPHTTPError as e:
+                exception = e
+                break
+            except (KeyError, IndexError) as e:
+                exception = e
+                if attempt == self.config['retries']:
+                    continue
+                await asyncio.sleep(backoff)
+                backoff = backoff ** 2
+
+        msg = f'Could not get external IP for {resource_name}: {exception}'
+        raise exceptions.GCPGordonError(msg)
+
+    async def _create_A_rrecord(self, data):
+        external_ip = await self._poll_for_ext_ip(data['resourceName'])
+        dns_zone = self.config['dns_zone']
+        hostname = data['resourceName'].split('/')[-1]
+
+        name = '.'.join([hostname, dns_zone])
+        logging.debug(f'Creating A record: {name} -> {external_ip}')
+        return {
+            'name': name,
+            'rrdatas': [external_ip],
+            'type': 'A'
```

Comment:
Hm, I set it up so that it's part of the enricher config/workflow - could you take a look and see if it makes sense this way? This would keep event message changes to just one plugin. What do you think?